tools/benchmark-cli/src/main/java/org/logstash/benchmark/cli/Main.java

```diff
@@ -0,0 +1,152 @@
+            ).withRequiredArg().ofType(String.class).forHelp();
+        final OptionSpec<String> local = localbuilder.requiredUnless(
+            UserInput.DISTRIBUTION_VERSION_PARAM, UserInput.GIT_VERSION_PARAM)
+            .availableUnless(UserInput.DISTRIBUTION_VERSION_PARAM, UserInput.GIT_VERSION_PARAM)
+            .withRequiredArg().ofType(String.class).forHelp();
+        final OptionSpec<String> testcase = parser.accepts(
+            UserInput.TEST_CASE_PARAM,
+            "Current available test cases are 'baseline' and 'apache'."
+        ).withRequiredArg().ofType(String.class).defaultsTo("baseline").forHelp();
+        final OptionSpec<File> pwd = parser.accepts(
+            UserInput.WORKING_DIRECTORY_PARAM,
+            "Working directory to store cached files in."
+        ).withRequiredArg().ofType(File.class).defaultsTo(UserInput.WORKING_DIRECTORY_DEFAULT)
+            .forHelp();
+        final OptionSet options;
+        try {
+            options = parser.parse(args);
+        } catch (final OptionException ex) {
+            parser.printHelpOn(System.out);
+            throw ex;
+        }
+        final LsVersionType type;
+        final String version;
+        if (options.has(distribution)) {
+            type = LsVersionType.DISTRIBUTION;
+            version = options.valueOf(distribution);
+        } else if (options.has(git)) {
+            type = LsVersionType.GIT;
+            version = options.valueOf(git);
+        } else {
+            type = LsVersionType.LOCAL;
+            version = options.valueOf(local);
+        }
+        execute(
+            new UserOutput(System.out), loadSettings(), options.valueOf(testcase),
+            options.valueOf(pwd).toPath(), version, type
+        );
+    }
+
+    public static void execute(final UserOutput output, final Properties settings,
+        final String test, final Path cwd, final String version, final LsVersionType type)
+        throws Exception {
+        output.printBanner();
+        output.printLine();
+        output.green(String.format("Benchmarking Version: %s", version));
+        output.green(String.format("Running Test Case: %s", test));
+        output.printLine();
+        Files.createDirectories(cwd);
+        final LogstashInstallation logstash = LsBenchLsSetup.setupLS(
+            cwd.toAbsolutePath().toString(), version, type,
+            JRubyInstallation.bootstrapJruby(cwd, JRubyInstallation.JRUBY_DEFAULT_VERSION)
+        );
+        final Case testcase;
+        if ("baseline".equals(test)) {
+            testcase = new GeneratorToStdout(logstash);
+        } else if ("apache".equals(test)) {
+            testcase = new ApacheLogsComplex(logstash, cwd, settings);
+        } else {
+            throw new IllegalArgumentException(String.format("Unknown test case %s", test));
+        }
+        output.printStartTime();
+        output.green("Statistical Summary:\n");
```

Comment:
Jup this makes sense :) man we should have some progress bar ... not sure how to do this without `libcurses` but that could be in version `0.2` :P